src/Components/Samples/BlazorServerApp/Pages/FetchData.razor

```diff
@@ -3,7 +3,6 @@
 <PageTitle>Weather forecast</PageTitle>
 
 @using BlazorServerApp.Data
-@inject WeatherForecastService ForecastService
```

Comment:
I updated the sample as an example. We cannot update any of the built in components that already use `@inject` since it's a binary breaking change to not have the parameterless ctor. However, we could use this in newly added components - it works a lot nicer with nullable contexts. For razor files, this is a bit of a wash since the resulting code is much more verbose, while being slightly less magical.

Reply:
Would it be OK to undo the changes to `BlazorServerApp`? We generally try to keep this aligned with the default project template, and not let it accumulate variations for each new feature we implement, as that maximizes its relevance for bug investigations.